Packs/AWS-IAMIdentityCenter/Integrations/AWSIAMIdentityCenter/AWSIAMIdentityCenter.py

```diff
@@ -0,0 +1,519 @@
+            user_details = {
+                'UserName': user['UserName'],
+                'UserId': user['UserId'],
+                'Email': user['Emails'][0]['Value'],
+                'DisplayName': user['DisplayName']
+            }
+            userID = user['UserId']
+            data.append(user_details)
+    ec = {'AWS.IAM.IdentityCenter.Users': data}
+    human_readable = tableToMarkdown('AWS IAM Users', data, removeNull=True)
+    return_outputs(human_readable, ec)
+    return userID
+
+def get_user_by_email(args, client):  # pragma: no cover
+    data = []
+    emailArg = demisto.getArg('emailAddress')
+    response = client.list_users(
+        IdentityStoreId=f'{IDENTITYSTOREID}',
+    )
+    rep = json.dumps(response)
+    repJSON = json.loads(rep)
+    datas = repJSON.get('Users', [])
+    for da in datas:
+        for user in response['Users']:
+            userEmail = user['Emails'][0]['Value']
+            if userEmail == emailArg:
+                user_details = {
+                    'UserName': user['UserName'],
+                    'UserId': user['UserId'],
+                    'Email': user['Emails'][0]['Value'],
+                    'DisplayName': user['DisplayName']
+                }
+                userID = user['UserId']
+                data.append(user_details)
+    ec = {'AWS.IAM.IdentityCenter.Users': data}
+    human_readable = tableToMarkdown('AWS IAM Users ', data, removeNull=True)
+    return_outputs(human_readable, ec)
+    return userID
+
+
+def list_users(args, client):  # pragma: no cover
+    data = []
+    response = client.list_users(
+        IdentityStoreId=f'{IDENTITYSTOREID}',
+    )
+    rep = json.dumps(response)
+    repJSON = json.loads(rep)
+    datas = repJSON.get('Users', [])
+    for da in datas:
+        for user in response['Users']:
+            user_details = {
+                'UserName': user['UserName'],
+                'UserId': user['UserId']
+            }
+            data.append(user_details)
+    ec = {'AWS.IAM.IdentityCenter.Users': data}
+    human_readable = tableToMarkdown('AWS IAM Identity Center Users', data, removeNull=True)
+    return_outputs(human_readable, ec)
+
+#def update_user(args, client):  # pragma: no cover
+#    kwargs = {'UserName': args.get('oldUserName')}
+#    if args.get('newUserName'):
+#        kwargs.update({'NewUserName': args.get('newUserName')})
+#    if args.get('newPath'):
+#        kwargs.update({'NewPath': args.get('newPath')})
+#
+#    response = client.update_user(**kwargs)
+#    if response['ResponseMetadata']['HTTPStatusCode'] == 200:
+#        demisto.results(
+#            "Changed UserName {0} To: {1}".format(args.get('oldUserName'), args.get('newUserName')))
+
+
+#def delete_user(args, client):  # pragma: no cover
+#    response = client.delete_user(UserName=args.get('userName'))
+#    if response['ResponseMetadata']['HTTPStatusCode'] == 200:
+#        demisto.results('The User {0} has been deleted'.format(args.get('userName')))
```

Comment:
Are these needed? If not, please delete them

Reply:
Done